### PR TITLE
Box2D - Basic ready for use (clean code and speed optimized)

### PR DESCRIPTION
**Pull request by aismann**

### CODE REVIEW

**Comment by aismann** [comment]
DebugDraw works now: I belive a big feature for using Box2D in an own project.

File: extensions/physics-nodes/CCPhysicsDebugNodeBox2D.h [hunk 1]
```diff
@@ -35,6 +35,66 @@ struct b2AABB;
 
 extern cocos2d::DrawNode* drawBox2D;
 
+/// Test settings. Some can be controlled in the GUI.
+struct Settings
```
Comment (halx99) [inline extensions/physics-nodes/CCPhysicsDebugNodeBox2D.h:39]: Name style suggetion: Settings to Box2dDrawSettings and put to namespace cocos2d::extensions maybe better

File: extensions/physics-nodes/CCPhysicsDebugNodeBox2D.h [hunk 2]
```diff
@@ -35,6 +35,66 @@ struct b2AABB;
...
+    int32 drawProfile;
+    int32 enableWarmStarting;
+    int32 enableContinuous;
+    int32 enableSubStepping;
+    int32 enableSleep;
+    int32 pause;
+    int32 singleStep;
+};
+
+struct ContactPoint
```
Comment (halx99) [inline extensions/physics-nodes/CCPhysicsDebugNodeBox2D.h:87]: put to namespace cocos2d::extensions maybe better

Comment (aismann) [reply extensions/physics-nodes/CCPhysicsDebugNodeBox2D.h:87]: I think Settings will be removed in the next clean up.
For the moment I improve/clean up the Box2D - TestBed.

Can you accept the pull request at the moment?

Comment (halx99) [reply extensions/physics-nodes/CCPhysicsDebugNodeBox2D.h:87]: Accepted

**Pull request merged**